### PR TITLE
added ability to clone a polarion testrun given a testrun id

### PR DESCRIPTION
**Pull request by pruan-rht**
```
./tools/polarshift.rb clone-run 20191205-1157
Source run id '20191205-1157' cloned to new run id: '20191207-0022'
```

### CODE REVIEW

**Comment by pruan-rht** [comment]
@akostadinov PTAL.

File: tools/polarshift.rb [hunk 1]
```diff
@@ -188,6 +188,23 @@ def run
...
+          "e.g. tools/polarshift.rb clone-run my_run_id"
+        c.action do |args, options|
+          setup_global_opts(options)
+
+          if args.size != 1
+            raise "command expects exactly one parameter being the test run id"
+          end
+
+          test_run_id = args.first
+          result = polarshift.get_run_smart(project, test_run_id)
```
Comment (akostadinov) [inline tools/polarshift.rb]: better move this into `clone_run` method

File: tools/polarshift.rb [hunk 2]
```diff
@@ -379,6 +396,60 @@ def setup_global_opts(options)
...
     end
+
+    # given a query_result from a call to get_smart_run()
+    # @return Array of testcase IDs
+    def extract_test_case_ids(query_result)
+      query_result['records']['TestRecord'].map { |tr| tr['test_case']['id'] }
+    end
+
+    # given a query_result Custom object (Array)
+    # @return Hash to be constructed
```
Comment (akostadinov) [inline tools/polarshift.rb]: could you clarify this? Maybe provide a simple example in comment?
Other than these two, lg

Comment (pruan-rht) [reply tools/polarshift.rb]: @akostadinov ok, updated.... PTAL again.  Thanks!

**Comment by pruan-rht** [comment]
@akostadinov PTAL again...Thanks!

File: tools/polarshift.rb [hunk 3]
```diff
@@ -379,6 +395,93 @@ def setup_global_opts(options)
...
+      return custom_fields
+    end
+
+    # given a query_result
+    # @return a YAML string that will be feed into create_run to clone a past
+    #         polarion run
+    # The YAML basically contains 3 fields
+    # 1.   "run_title":  "Clone of xyz run"
+    # 2.   "case_query": "<SQL_STATEMENT>"
+    # 3.   "custom_fields": <CONSTRUCTED_FROM_call_to_get_smart_run_method>
```
Comment (akostadinov) [inline tools/polarshift.rb]: I think info above is outdated. Please update it.

Other than that looks nice!

Comment (pruan-rht) [reply tools/polarshift.rb]: @akostadinov comment updated,  PTAL again.

**Pull request merged**